Update docs for networking Runtime Configuration Options (#21197)
dotnet/docs: docs/core/run-time-config/networking.md

- Configures whether support for the HTTP/2 protocol is enabled.

- Introduced in .NET Core 3.0.

> - In .NET Core 3.0: If you omit this setting, support for the HTTP/2 protocol is disabled. This is equivalent to setting the value to `false`.

Reviewer: ```suggestion
- .NET Core 3.0 only: If you omit this setting, support for the HTTP/2 protocol is disabled. This is equivalent to setting the value to `false`.
```

> - Since .NET Core 3.1: If you omit this setting, support for the HTTP/2 protocol is enabled. This is equivalent to setting the value to `true`.

Reviewer: ```suggestion
- .NET Core 3.1 and .NET 5+: If you omit this setting, support for the HTTP/2 protocol is enabled. This is equivalent to setting the value to `true`.
```

> [!NOTE]
> Starting in .NET 5, the `System.Net.Http.UseSocketsHttpHandler` setting is no longer available.

> ## Allow sending Latin1 headers in .NET Core 3.1

Reviewer: ```suggestion
## Latin1 headers (.NET Core 3.1 only)
```
